com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -221,12 +229,12 @@ public static UnityRLInitParameters ToUnityRLInitParameters(this UnityRLInitiali
         }
 
         #region AgentAction
-        public static List<float[]> ToAgentActionList(this UnityRLInputProto.Types.ListAgentActionProto proto)
+        public static List<ActionBuffers> ToAgentActionList(this UnityRLInputProto.Types.ListAgentActionProto proto)
         {
-            var agentActions = new List<float[]>(proto.Value.Count);
+            var agentActions = new List<ActionBuffers>(proto.Value.Count);
             foreach (var ap in proto.Value)
             {
-                agentActions.Add(ap.VectorActions.ToArray());
+                agentActions.Add(new ActionBuffers(ap.ContinuousActions.ToArray(), ap.DiscreteActions.ToArray()));
```

Comment:
Not quite sure what type `ap` is here, but can we add a ToActionBuffers() extension method to it?

Reply:
It is auto-generated `AgentActionProto` class so maybe not

Reply:
extension method == "this" modifier in argument, so something like
```csharp
public static ActionBuffers ToActionBuffers(this UnityRLInputProto.Types.AgentActionProto proto) 
{
    return new ActionBuffers(proto.ContinuousActions.ToArray(), proto.DiscreteActions.ToArray());
}

...
agentActions.Add(ap.ToActionBuffers());
...
```

Reply:
not a big deal but it's just a bit cleaner